src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +34,18 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * Generates the ID of an "instanceof" definition.
+     *
+     * @param string $id
+     * @param string $type
+     * @param string $file
+     *
+     * @return string
+     */
+    protected function generateInstanceofDefinitionId($id, $type, $file)
+    {
+        return sprintf('%s_%s_%s', $id, $type, hash('sha256', $file));
```

Comment:
generated ids should start with a number so that we don't confuse them with classes (maybe not applicable here, but still a good default practice to me).